docs/source/quickstart.rst

```diff
@@ -1,7 +1,68 @@
 Quickstart
 ==========
+Instantiate a new client from a configuration file
+--------------------------------------------------
+Use this code to instantiate a client in order to execute API calls to Cognite Data Fusion (CDF) using a configuration file.
+
+.. note::
+    How you read in the configuration file is up to you as the :ref:`CogniteClient <class_client_CogniteClient>` load method only
+    accepts a dictionary. So for the purposes of this example, we will use the yaml library to read in a yaml file and
+    substitute environment variables in the file string to ensure that sensitive information is not stored in the file.
+
+See :ref:`CogniteClient <class_client_CogniteClient>`, :ref:`ClientConfig <class_client_ClientConfig>`,
+:ref:`GlobalConfig <class_client_GlobalConfig>`, and :ref:`credential_providers:Credential Providers`
+for more information on the configuration options.
+
+.. code:: yaml
+
+    # cognite-sdk-config.yaml
+    client:
+      project: "my-project"
+      client_name: "my-special-client"
+      base_url: "https://${MY_CLUSTER}.cognitedata.com"
+      credentials:
+        client_credentials:
+          token_url: "https://login.microsoftonline.com/${MY_TENANT_ID}/oauth2/v2.0/token"
+          client_id: "${MY_CLIENT_ID}"
+          client_secret: "${MY_CLIENT_SECRET}"
+          scopes: ["https://api.cognitedata.com/.default"]
+    global:
+      max_retries: 10
+      max_retry_backoff: 10
+
+.. code:: python
+
+    import os
+    from pathlib import Path
+    from string import Template
+
+    import yaml
+
+    from cognite.client import CogniteClient
+    from cognite.client.config import GlobalConfig
+
+    file_path = Path("cognite-sdk-config.yaml")
+
+    # Read in yaml file and substitute environment variables in the file string
+    with file_path.open("r") as file_raw:
+        env_sub_template = Template(file_raw.read())
+    try:
+        file_env_parsed = env_sub_template.substitute(dict(os.environ))
+    except (KeyError, ValueError) as e:
+        raise ValueError(f"Error substituting environment variable: {e}")
+
+    # Load yaml file string into a dictionary
+    cognite_config = yaml.safe_load(file_env_parsed)
```

Comment:
I'd suggest we remove this step

Reply:
if I want to make a GlobalConfig and CogniteClient in the example then I'd have to do this so I can get the key for each. but maybe easiest to take an offline chat about this whole example since there were other comments too

Reply:
Oh, right 👍